opm/grid/cpgrid/CpGridUtilities.cpp

```diff
@@ -202,21 +199,25 @@ void setPillarCoordinates(int i, int j, int nx,
     // - positionIdx / 2 determines the position at the y-axis (0 for j, 1 for j+1).
     // - positionIdx % 2 determines the position at the x-axis (0 for i, 1 for i+1).
 
-    const int pillar = ((j + positionIdx / 2) * (nx + 1)) + (i + positionIdx % 2);
+    const int pillar = ((j + positionIdx / 2) *6* (nx + 1)) + 6*(i + positionIdx % 2);
 
     // Top pillar's COORD values
     const auto& top_point = topElem.subEntity<3>(topCorner).geometry().center();
-    std::copy(top_point.begin(), top_point.end(), lgrCOORD[pillar].begin());
+    lgrCOORD[pillar] = top_point[0];
+    lgrCOORD[pillar+1] = top_point[1];
+    lgrCOORD[pillar+2] = top_point[2];
```

Comment:
Why not stick with copy? 
```suggestion
    std::copy(top_point.begin(), top_point.end(), lgrCOORD.begin()+pillar);
```